### ISSUE TITLE
False positive with `VoidMissingNullable` 

### ISSUE THREAD

**Issue by Stephan202** [google/error-prone#2679]
Consider the following code:
```java
import com.sun.source.util.TreeScanner;
import org.checkerframework.checker.nullness.qual.Nullable;

class Example {
  void methodWithAnonymousClass() {
    new TreeScanner<@Nullable Void, @Nullable Void>() {};
  }
}
```

Compile the code as follows:
```sh
javac \
  -XDcompilePolicy=simple \
  -processorpath dataflow-errorprone-3.15.0.jar:error_prone_core-2.10.0-with-dependencies.jar \
  -cp checker-qual-3.19.0.jar \
  '-Xplugin:ErrorProne -XepDisableAllChecks -Xep:VoidMissingNullable:WARN' \
  Example.java
```

This yields the following warnings:
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.errorprone.util.FindIdentifiers (file:/home/sschroevers/tmp/repro-2/error_prone_core-2.10.0-with-dependencies.jar) to method com.sun.tools.javac.comp.Resolve.findIdent(com.sun.tools.javac.comp.Env,com.sun.tools.javac.util.Name,com.sun.tools.javac.code.Kinds$KindSelector)
WARNING: Please consider reporting this to the maintainers of com.google.errorprone.util.FindIdentifiers
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Example.java:6: warning: [VoidMissingNullable] The type Void is not annotated @Nullable
    new TreeScanner<@Nullable Void, @Nullable Void>() {};
                    ^
    (see https://errorprone.info/bugpattern/VoidMissingNullable)
  Did you mean 'new TreeScanner<@Nullable @Nullable Void, @Nullable Void>() {};'?
Example.java:6: warning: [VoidMissingNullable] The type Void is not annotated @Nullable
    new TreeScanner<@Nullable Void, @Nullable Void>() {};
                                    ^
    (see https://errorprone.info/bugpattern/VoidMissingNullable)
  Did you mean 'new TreeScanner<@Nullable Void, @Nullable @Nullable Void>() {};'?
Example.java:6: warning: [VoidMissingNullable] The type Void is not annotated @Nullable
    new TreeScanner<@Nullable Void, @Nullable Void>() {};
                    ^
    (see https://errorprone.info/bugpattern/VoidMissingNullable)
  Did you mean 'new TreeScanner<@Nullable @Nullable Void, @Nullable Void>() {};'?
Example.java:6: warning: [VoidMissingNullable] The type Void is not annotated @Nullable
    new TreeScanner<@Nullable Void, @Nullable Void>() {};
                                    ^
    (see https://errorprone.info/bugpattern/VoidMissingNullable)
  Did you mean 'new TreeScanner<@Nullable Void, @Nullable @Nullable Void>() {};'?
4 warnings
```

Two things seem to be going on here:
1. There are false-positives. I attached a debugger and noticed that the type passed [here](https://github.com/google/error-prone/blob/f10bba5891d9c7bf8588792fac04afbc238478fd/core/src/main/java/com/google/errorprone/bugpatterns/nullness/VoidMissingNullable.java#L115) lacks the relevant `@Nullable` annotation mirror. Didn't investigate too deeply, but the fix isn't obvious to me at this moment. :thinking:
2. The violations are reported twice.

CC @cpovirk

NB: [`org.checkerframework.checker.nullness.qual.Nullable`](https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/Nullable.html) is annotated `@DefaultFor(types=java.lang.Void.class)`, so I guess users of that annotation maybe shouldn't enable `VoidMissingNullable`. But IIUC the report also applies to other `@Nullable` variants.

**Comment by cpovirk** [comment]
Interesting, thanks. Sorry for the trouble.

My first thought when I saw the bug title was that this was an issue we recently discovered with inner classes: See https://github.com/google/error-prone/commit/ce87ca1c7bb005371837b82ffa69041dd8a356e5. But that doesn't explain a problem with `Void`, nor does it explain the duplication.

(Now that I think about it, we might have seen a problem with `Void` at one point, too......)

I have a thought on how to fix the inner-class problem, and there's a _chance_ that it will take care of whatever is going wrong here. If not, I'm not sure how soon I'll have a look: As you note, Checker Framework users don't need to annotate `Void` in the first place, so the refactoring is mostly for users of Kotlin or other tools.

**Comment by Stephan202** [comment]
Tnx for the rapid fix @cpovirk! :rocket:
